### ISSUE TITLE
Slider shift when item clicked

### ISSUE THREAD

**Issue by lsps9150414** [archriss/react-native-snap-carousel#33]
The slider shift when item is clicked.
![2 -06-2017 17-16-05](https://cloud.githubusercontent.com/assets/5022617/22640995/f7d10e3c-ec8f-11e6-97a7-36f10cf139b1.gif)

Here's my code:
```javascript
const SLIDER_WIDTH = Dimensions.get('window').width;
const SLIDE_ITEM_MARGIN = 35;
const SLIDE_ITEM_WIDTH = SLIDER_WIDTH - (SLIDE_ITEM_MARGIN * 2);
```

```javascript
      <Carousel
        sliderWidth={SLIDER_WIDTH}
        itemWidth={SLIDE_ITEM_WIDTH}
        style={{ paddingHorizontal: 0 }}
      >
        {this.renderMedicationCards(this.props.medications)}
      </Carousel>
```

This does not happen when I set my `SLIDE_ITEM_MARGIN` to 0, which is setting `SLIDE_ITEM_WIDTH` = `SLIDER_WIDTH` = `Dimensions.get('window').width`

**Comment by bd-arc** [comment]
@lsps9150414 Hi! Thanks for providing an animated screenshot; that really helped clarifying the issue. What's happening there is that the ScrollView is scrolled past its bounds because you've overridden the carousel container's style with `style={{ paddingHorizontal: 0 }}`.

First, you need to make sure to use the `containerCustomStyle` prop whenever you want to style carousel's container.

Then, **you should never override the horizontal padding of both ScrollView's container and ScrollView's content container**. As you can see in the [source code](https://github.com/archriss/react-native-snap-carousel/blob/master/index.js#L442:L450), this is a critical value which allows for slide centering. If you need horizontal padding around the carrousel, I suggest that you wrap it inside a View component that you can style according to your needs.

Hope that helps! Let me know if this solves the issue for you.

**Comment by lsps9150414** [comment]
@bd-arc Thanks for the fast reply!

It seems to be some other reason causing the problem. I've removed any styles attached to `Carousel`. Not `style` nor `containerCustomStyle`. It still happens.
![2 -06-2017 23-12-06](https://cloud.githubusercontent.com/assets/5022617/22652585/f9851b1a-ecc1-11e6-902f-9bae66c47b8f.gif)

**Comment by bd-arc** [comment]
@lsps9150414 I still think that the issue probably has to do with a faulty padding's calculation. Something might have changed with recent versions of React Native. Which version are you using and on which OS are you testing?

This is working properly for me with versions 0.38 to 0.41 of RN, on both iOS and Android. So I'm confident that we can find a solution to your issue ;)

**Comment by lsps9150414** [comment]
It turns out to have something to do with `enableMomentum`. Using your example, when I set it to false same thing happens. I haven't got time to dig into the src code to figure out the problem.
![2 -07-2017 14-59-10](https://cloud.githubusercontent.com/assets/5022617/22681050/46dc9120-ed46-11e6-9ea3-5faa267a0cfd.gif)

**Comment by bd-arc** [comment]
@lsps9150414 This is really surprising as we are using it with `enableMomentum` equally set to `false` or `true` without any issue.

Can you tell me the version of React Native you are using as well as the OS on which you have tested the behavior ?

**Comment by lsps9150414** [comment]
The above demo is RN 0.38, on mac simulator iOS v 9.3.

**Comment by bd-arc** [comment]
@lsps9150414 Hi! I've tried the current example with the same configuration as yours (on both emulator and real device), and I wasn't able to reproduce the issue you're facing...

![preview](https://media.giphy.com/media/26xByqT1X8SlZiYow/giphy.gif)

Could you download [this version of the example](https://github.com/archriss/react-native-snap-carousel/tree/c115df61dc44aae97671c499b8edac6a8a3f2459/example), [freeze plugin's version](https://github.com/archriss/react-native-snap-carousel/blob/c115df61dc44aae97671c499b8edac6a8a3f2459/example/package.json#L12) to `1.4.0` and see if that does anything for you?

**Comment by bd-arc** [comment]
@lsps9150414 Would you mind trying with the latest version of the plugin and tell me if you can still reproduce the issue?

**Comment by bd-arc** [comment]
Closing as the issue should be resolved in version 2.1.1.
Feel free to reopen if needed.

**Comment by darkhorse-coder** [comment]
This issue is appearing again. when I click items (not first and end), getting left swift.

![Screen Capture on 2021-04-21 at 10-18-06](https://user-images.githubusercontent.com/32187178/115487278-0dae8280-a261-11eb-9ef1-bbd04381f599.gif)

below my code:

`
<Carousel
               ref={carouselRef}
               sliderWidth={width}
               data={pages}
               renderItem={renderPage}
               itemWidth={width - 56}
               onSnapToItem={onChangeSlide}
	       inactiveSlideScale={1}
	       pagingEnabled
	       firstItem={type - 1}
            />
`
